### ISSUE TITLE
Unable to use nullable types with injection parameters

### ISSUE THREAD

**Issue by 0x15e** [z4kn4fein/stashbox#49]
When I have a class with a nullable type in the constructor that needs an injection parameter, I get the following error trying to resolve it:

`ArgumentException: Expression of type 'System.Int32' cannot be used for constructor parameter of type 'System.Nullable`1[System.Int32]'
Parameter name: arguments[0]`

Here's the class:
```
    public interface IExample
    {
        int? ExampleProperty { get; }
    }

    public class ExampleClass : IExample
    {
        private readonly int? _exampleValue;

        public ExampleClass(int? exampleValue = null)
        {
            _exampleValue = exampleValue;
        }

        public int? ExampleProperty => _exampleValue;
    }
```

And here's the registration / configuration:

```
        public void ConfigureContainer(IStashboxContainer container)
        {
            container.Configure(config => config.WithOptionalAndDefaultValueInjection().WithCircularDependencyTracking());

            var someInt = (int?)123;

            container.Register<IExample, ExampleClass>(
                config => config.WithInjectionParameters(
                    new InjectionParameter
                    {
                        Name = "exampleValue",
                        Value = someInt
                    }));
        }
```

**Comment by z4kn4fein** [comment]
Thanks for pointing this out! I'll check it soon and publish the fix with the next release!

**Comment by 0x15e** [comment]
Awesome. Thanks!

**Comment by z4kn4fein** [comment]
I pushed a new version to my [pre-release](https://www.myget.org/feed/pcsajtai/package/nuget/Stashbox) feed which contains the fix for this issue, could you please check and confirm that it works as expected?
Thanks!

**Comment by z4kn4fein** [comment]
fixed in [v2.6.8](https://github.com/z4kn4fein/stashbox/releases/tag/2.6.8)

**Comment by 0x15e** [comment]
Sorry, I totally missed your message. I'm sure the fix is fine but I'll give it a test tomorrow.
